programs/ziti-edge-tunnel/log_utils.c

```diff
@@ -0,0 +1,308 @@
+        return false;
+    }
+    return true;
+}
+
+void close_log() {
+    if (ziti_tunneler_log != NULL) {
+        fclose(ziti_tunneler_log);
+        ziti_tunneler_log = NULL;
+    }
+    if (log_filename != NULL) {
+        free(log_filename);
+        log_filename = NULL;
+    }
+}
+
+void stop_log_check() {
+    if (log_flusher != NULL) {
+        uv_check_stop(log_flusher);
+        free(log_flusher->data);
+        free(log_flusher);
+    }
+}
+
+void rotate_log() {
+    close_log();
+
+    uv_timeval64_t file_time;
+    uv_gettimeofday(&file_time);
+#if _WIN32
+    _gmtime32_s(start_time, &file_time.tv_sec);
+#else
+    gmtime_r(&file_time.tv_sec, start_time);
+#endif
+    log_filename = get_log_filename();
+
+    open_log(log_filename);
+}
+
+void delete_older_logs(uv_loop_t *ziti_loop) {
+    char curr_path[FILENAME_MAX]; //create string buffer to hold path
+#if _WIN32
+    _getcwd( curr_path, FILENAME_MAX );
+#else
+    sprintf(curr_path, "%s", "/tmp");
+#endif
+
+    char log_path[FILENAME_MAX];
+    sprintf(log_path, "%s/logs", curr_path);
+
+    uv_fs_t fs;
+    int rc = uv_fs_scandir(ziti_loop, &fs, log_path, 0, NULL);
+    if (rc < 7) {
+        if (rc < 0) {
+            ZITI_LOG(ERROR, "failed to scan dir[%s]: %d/%s", log_path, rc, uv_strerror(rc));
+        } else {
+            ZITI_LOG(TRACE, "Log files count in [%s] is %d, not deleting log files.", log_path, rc);
+            uv_fs_req_cleanup(&fs);
+        }
+        return;
+    }
+
+    char **log_files = calloc(rc + 1 , sizeof(char *));
+    uv_dirent_t file;
+    int rotation_cnt = 0;
+    while (uv_fs_scandir_next(&fs, &file) == 0) {
+        ZITI_LOG(TRACE, "log file = %s %d", file.name, file.type);
+
+        if (file.type == UV_DIRENT_FILE) {
+            if (strncmp(file.name, log_filename_base, strlen(log_filename_base)) == 0) {
+                log_files[rotation_cnt] = strdup(file.name);
+                rotation_cnt++;
+            }
+        }
+    }
+
+    char logpath[FILENAME_MAX];
+    sprintf(logpath, "%s/logs", curr_path);
+    int rotation_index = rotation_cnt;
+    while (rotation_index > rotation_count) {
+        char *old_log = NULL;
+        int old_idx = -1;
+
+        for(int idx =0; idx < rotation_cnt; idx++) {
+            if (old_log == NULL && log_files[idx]) {
+                old_log = log_files[idx];
+                old_idx = idx;
+                continue;
+            }
+            if (old_log == NULL) {
+                continue;
+            }
+            if (strcmp(old_log, log_files[idx]) > 0 ) {
+                old_log = log_files[idx];
+                old_idx = idx;
+            }
+        }
+        if (old_log != NULL) {
+            char logfile_to_delete[MAXPATHLEN];
+            sprintf(logfile_to_delete, "%s/%s", logpath, old_log);
+            ZITI_LOG(INFO, "Deleting old log file %s", logfile_to_delete);
+            remove(logfile_to_delete);
+            rotation_index--;
+            free (old_log);
```

Comment:
Should old_log be set to NULL here? I'm concerned about a crash on line 283 if the pointer isn't null'ed after freeing, and the rotation index is still greater than rotation count.